Inspicio/Controllers/ImagesController.cs

```diff
@@ -107,29 +108,51 @@ public class DataToBody
                 return NotFound();
             }
 
+            // Jack Lloyd 
+            // Fetch the image by the id pass in '/5'
             var image = await _context.Images.SingleOrDefaultAsync(m => m.ImageID == id);
             if (image == null)
             {
                 return NotFound();
             }
 
-            DataToBody DataToBody = new DataToBody();
-            DataToBody.Image = image;
+            // Jack Lloyd 
+            // Create a new FullReview object
+            // This will be passed into the view as the model.
+            FullReview FullReview = new FullReview();
+            FullReview.Image = image;
 
+            // Jack Lloyd 
+            // Fetch all the comments 
+            // This may not be the best way to do this...
             var allcomments = _context.Comments;
-            List<Comment> comments = new List<Comment>();
+
+            // Jack Lloyd 
+            // The FullReview comments holds a tuple holding one comment and one username.
+            List<Tuple<Comment, String>> comments = new List<Tuple<Comment, String>>();
 
             foreach (Comment comment in allcomments)
             {
+                // Jack Lloyd 
+                // We only care about the comments for this image
                 if (comment.ImageId == id)
                 {
-                    comments.Add(comment);
+                    // Jack Lloyd 
+                    // add it and query the user table for the profilename.
+                    comments.Add( new Tuple<Comment, String>(comment, _context.Users.SingleOrDefault(u => u.Id == comment.OwnerId).ProfileName));
```

Comment:
You could ask the database for less data, seeing as you're only interested in the `ProfileName` with something like
`comments.Add( new Tuple<Comment, String>(comment, _context.Users.Where(x => x.Id == comment.OwnerId).Select(x => x.ProfileName).SingleOrDefault()));`

You could also reduce the number of calls to the DB by getting the profile names at the same time as the comments at the same time, but that might not be worth the effort (at least not yet while stuff is in flux)